documentation/PreferencesServer.md

```diff
@@ -350,6 +350,434 @@ There are two important things to note here:
+
+Example returned item:
+
+```json
+{
+    "prefsSafe": {
+        "id": "prefsSafe-alice",
+        "type": "prefsSafe",
+        "schemaVersion": "0.1",
+        "prefsSafeType": "user",
+        "name": "alice",
+        "email": null,
+        "preferences": {
+            "flat": {
+                "name": "Alice",
+                "contexts": {
+                    "gpii-default": {
+                        "name": "Default preferences",
+                        "preferences": {
+                            "http://registry.gpii.net/applications/com.microsoft.windows.onscreenKeyboard": {}
+                        }
+                    }
+                }
+            }
+        },
+        "timestampCreated": "2019-01-14T23:48:03.221Z",
+        "timestampUpdated": null
+    },
+    "keys": [
+        {
+            "id": "8f3085a7-b65b-4648-9a78-8ac7de766997",
+            "type": "gpiiCloudSafeCredential",
+            "schemaVersion": "0.1",
+            "prefsSafeId": "prefsSafe-alice",
+            "gpiiExpressUserId": "org.couch.db.user:alice"
+        },
+        {
+            "id": "57A68E84-03A9-4ADD-9365-11C75E4F1B0E",
+            "type": "gpiiKey",
+            "schemaVersion": "0.1",
+            "prefsSafeId": "prefsSafe-alice",
+            "prefsSetId": "gpii-default",
+            "revoked": false,
+            "revokedReason": null,
+            "timestampCreated": "2017-12-14T19:55:11.640Z",
+            "timestampUpdated": null
+        }
+    ]
+}
+```
+
+If the safe with the specified `prefsSafeId` does not exist, the following error payload will be returned:
+
+```json
+{
+    "isError": true,
+    "errorCode": "GPII_ERR_NO_PREFSSAFE",
+    "message": "Missing prefssafe"
+}
+```
+
+### POST /prefssafe
+
+Creates a new preferences safe in the system. Takes a full preferences safe JSON payload, albiet without a `id`.
+The returned payload will include the entire preferences safe, including the updated `timestampCreated` field,
+and a newly provisioned `id`.
+
+#### Example POST
+
+URL: `http://preferences.gpii.net/prefssafe`
+
+The POST payload should be a full preferences safe without an `id`:
+
+```json
+{
+    "type": "prefsSafe",
+    "schemaVersion": "0.1",
+    "prefsSafeType": "user",
+    "name": "Steve",
+    "email": null,
+    "preferences": {
+        "flat": {
+            "name": "bit of stuff",
+            "contexts": {
+                "gpii-default": {
+                    "name": "Default preferences",
+                    "preferences": {
+                        "http://registry.gpii.net/common/onScreenKeyboard/enabled": false
+                    },
+                    "metadata": []
+                }
+            },
+            "metadata": []
+        }
+    }
+}
+```
+
+A successful return payload will consist of the same payload with the addition of timestamps and a
+freshly generated `id` field.  In the event of an error, a payload similar to the following will be returned

```

Comment:
"In the event of an error"... what is this error? Could you describe actual errors like what you did above at line 461 rather than using an fabricated error code ""GPII_ERR_APPLICABLE_MESSAGE"? Thanks.

This comment applies to other places in this doc.

Reply:
I went back through each `isError` message doc in the markdown. In situations where I'm actually doing specific checks I've put in the potential items to look for.  The remainders of them are actual general error catches for anything potentially unexpected, which is why they are general error messages, but do include the message.